docs/source/security/cors.mdx

```diff
@@ -0,0 +1,155 @@
+
+The [_origin_](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy#definition_of_an_origin) of a piece of web content consists of that content's domain, protocol, and port. The [same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) (SOP) is a security mechanism that restricts scripts on one origin from interacting with resources from another origin. This means that scripts on websites can interact with resources from the same origin without jumping through any extra hoops.
+
+If two URLs differ in their domain, protocol, or port, then those URLs come from two different origins:
+```bash
+# Same origin
+http://example.com:8080/ <==> http://example.com:8080/
+
+# Different origin (difference in domain, protocol, and port)
+http://example.com:8080/ =X= https://example1.com:8081/
+```
+
+However, as we all know, the internet is an exciting place full of resources that can make websites better (importing images, extra fonts, making API calls, etc.).  Developers needed a new protocol to relax SOP and safely _share_ resources across different origins.
+
+Cross-Origin Resource Sharing is the mechanism that allows a web page to  _share_ resources across different origins. CORS provides an extra layer of protection by enabling servers and clients to define HTTP headers that specify _which_ external clients' scripts can access their resources.
+
+Note that both SOP and CORS are related to [browser security](https://developer.mozilla.org/en-US/docs/Web/Security#security-related_glossary_terms); neither prevents other kinds of software from requesting resources from your server.
+
+## Choosing CORS options for your project
+
+When thinking about configuring CORS for your application, there are two main options you might want to change: the origins that can access your server's resources and whether your server accepts user credentials (i.e., cookies) with requests.
+
+### Specifying origins
+
+CORS uses [specific HTTP request headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#the_http_response_headers) as part of its protocol, including [`Access-Control-Allow-Origin`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin). The `Access-Control-Allow-Origin` header (ACAO) enables a server to dictate which origins can use scripts to access that server's resources.
+
+Depending on what you are building, the origins you specify in your CORS configuration might need to change when you are ready to deploy your application.
+
+#### ⚠️ Applications on private networks
+If your browser is running your API on a private network (i.e., not on the public internet), **the default CORS behavior of Apollo Server is insecure**. We strongly recommend [specifying which origins](#configuring-cors-options-for-apollo-server) can access your server's resources.
```

Comment:
```suggestion
If your browser is running your API on a private network (i.e., not on the public internet) and relies only on the privacy of the network for security, **the default CORS behavior of Apollo Server is insecure**. We strongly recommend [specifying which origins](#configuring-cors-options-for-apollo-server) can access your server's resources.
```

Reply:
I think it would also be helpful to get something here linking to https://www.apollographql.com/docs/federation/subgraphs/#securing-your-subgraphs and noting that Federation subgraphs generally should not be directly accessed by clients and if you've chosen to use network separation to implement this (rather than some sort of authentication by the gateway) then this is relevant for these subgraphs.